projects/js-packages/publicize-components/src/components/share-buttons/usePrepareUrl.ts

```diff
@@ -0,0 +1,37 @@
+import { useSelect } from '@wordpress/data';
+import { useCallback } from 'react';
+
+/**
+ * Prepares the URL to share.
+ *
+ * @returns {(urlWithPlaceholders: string) => string} A function that accepts a URL with placeholders and returns a URL with the placeholders replaced.
+ */
+export function usePrepareUrl() {
+	const getEditedPostAttribute = useSelect( select => {
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		return ( select( 'core/editor' ) as any ).getEditedPostAttribute satisfies (
+			attribute: string
+		) => unknown;
+	}, [] );
+
+	return useCallback(
+		( urlWithPlaceholders: string ) => {
+			// TODO update this to use custom message
```

Comment:
Should we retrieve the custom message in this change too, or will that be a separate task?

Reply:
We can do that in a separate task.

Reply:
I have done that in this PR. Thanks.